srml/democracy/src/lib.rs

```diff
@@ -154,6 +155,22 @@ decl_module! {
 				runtime_io::print(e);
 			}
 		}
+
+		/// Delegate vote.
+		pub fn delegate(origin, ref_index: ReferendumIndex, to: T::AccountId, lock_periods: LockPeriods) -> Result {
+			let who = ensure_signed(origin)?;
```

Comment:
This should ensure that `who != to`, maybe we should also check for cycles unless we prevent them with some other means.

Reply:
Should also check that it's not voting already.

Reply:
I added a check for cycles, but delegate is not more O(1) :-(.

Currently, delegation is working like a default that can be overwritten by voting. If the user votes, the delegation edge is not taken into account and then the vote is not counted twice. Not sure if this is right behavior, I can change if you want, but then I think I will need to iterate over the votes when delegating.

Reply:
that makes sense.